### ISSUE TITLE
Tag onClick does not fire with spacebar or enter press when focused

### ISSUE THREAD

**Issue by evansjohnson** [palantir/blueprint#4782]
#### Code Sandbox
https://codesandbox.io/s/palantir-blueprint-forked-s0reo

#### Steps to reproduce
1. click tag, observe onClick handler works
2. use tab to navigate to tag, onClick does not fire with spacebar or enter

`Button` included to show expected behavior

#### Actual behavior
`onClick` of `Tag` does not fire with spacebar or enter press when focused

#### Expected behavior
spacebar/enter should trigger `onClick` of `Tag`

#### Possible solution
if `Tag` has `onClick` defined, render as a button but with the `Tag` styles

**Comment by adidahiya** [comment]
I'm not sure this is expected behavior for tags. If you really want something so interactive that it can be focussed and clicked via the keyboard, you probably want a minimal or outlined button instead. I don't think your requested feature exists in other design toolkits like Bootstrap or Ant (tags are not even focussable by default in Ant).

**Comment by evansjohnson** [comment]
> I'm not sure this is expected behavior for tags. If you really want something so interactive that it can be focussed and clicked via the keyboard, you probably want a minimal or outlined button instead. I don't think your requested feature exists in other design toolkits like Bootstrap or Ant (tags are not even focussable by default in Ant).

"so interactive that ..." seems to indicate levels of interactivity, I think it's a binary thing. Either an element is interactive and it should be accessible by keyboard users, or it's not interactive and doesn't need to receive focus/be triggered by keyboard.

I'd support `Tag`s not accepting the `onClick` prop/no interactive style but that sounds like a large breaking change, but don't think that's a reason for `Tag`s to stay in a state of partial support for interactivity.

I don't need an immediate solution but thought raising this would be helpful to track accessibility of Blueprint components, particularly if the issue sees other activity. Perhaps a "Domain: accessibility" label is appropriate here.
